app/models/user_stats.rb

```diff
@@ -10,7 +10,39 @@ def self.exp_for(user)
     self.stats_for(user).exp
   end
 
+  def activity(date_range = nil)
+    {
+        exercises: {
+            solved_count: organization_exercises
+                              .joins(:assignments)
+                              .where(assignments: { top_submission_status: [:passed, :skipped], submitter: user }.merge_if_present(:submitted_at, date_range))
+                              .count,
+            count: organization_exercises.count},
+
+        messages: {
+            count: messages_in_discussions(date_range).count,
+            approved: messages_in_discussions(date_range).where(approved: true).count}
+    }
+  end
+
   def add_exp!(points)
     self.exp += points
   end
+
+  private
+
+  def messages_in_discussions(date_range = nil)
+    Message.joins(:discussion)
+        .where({sender: user.uid, discussions: { organization: organization }}.merge_if_present(:date, date_range))
+  end
+
+  def organization_exercises
+    organization.book.exercises
+  end
+end
+
+class Hash
+  def merge_if_present(key, value)
```

Comment:
I think this is a good approach but I'm worried about cases like {do_something_bad: false}, which would be filtered and I don't think we want that :confused:

Reply:
In any case, let's please add extension methods in the appropriate directory in order to keep things organized and avoid problems with rails lazy-loading. 

See https://github.com/mumuki/mumuki-domain/blob/master/lib/mumuki/domain/extensions/hash.rb